### PR TITLE
A few fixes for 0.6

### PR DESCRIPTION
**Pull request by yuyichao**
* `num` and `den` rename
* `takebuf_array` rename
* Fix `$` in the test

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1512,3 +1512,13 @@ end
...
 @test 1 ⊻ 5 == 4
+
+# julia#19246
+@test numerator(1//2) === 1
+@test denominator(1//2) === 2
+
+# julia#19088
+let io = IOBuffer()
+    write(io, "aaa")
+    @test take!(io) == UInt8['a', 'a', 'a']
```
Comment (yuyichao) [inline test/runtests.jl]: Sure. It's actually tested twice above already.

Comment (nalimilan) [inline test/runtests.jl]: Yes, but it doesn't hurt to do that explicitly here too.

File: README.md [hunk 2]
```diff
@@ -126,6 +126,10 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `$` is now `xor` or `⊻` [#18977](https://github.com/JuliaLang/julia/pull/18977).
 
+* `num` and `den` are now `numerator` and `denominator` [#19246](https://github.com/JuliaLang/julia/pull/19246).
+
+* `takebuf_array` is now a method of `take!`. `takebuf_string(io)` becomes `String(take!(io))` [#19088](https://github.com/JuliaLang/julia/pull/19088).
```
Comment (stevengj) [inline README.md]: Shouldn't this be `Compat.UTF8String(take!(io))` thanks to the `Compat.String` mess?

Comment (yuyichao) [inline README.md]: No, I believe `takebuf_string` on 0.4 is actually type unstable. It can return either `ASCIIString` or `UTF8String`.

Comment (stevengj) [inline README.md]: Then we need to define a ByteString(::Array{UInt8}) method

Comment (yuyichao) [inline README.md]: [Yes?](https://github.com/JuliaLang/Compat.jl/blob/master/src/Compat.jl#L1516)

Comment (TotalVerb) [inline README.md]: Don't we already?

Comment (yuyichao) [inline README.md]: Is there anything that's broken? This won't work before the specific 0.4 commit but Compat.jl already require 0.4.0 now. Travis tests also seems to be passing on 0.4?

**Comment by stevengj** [comment]
Right, sorry, we already have it

**Pull request merged**